### ISSUE TITLE
The recent changes may wrongly call async functions

### ISSUE THREAD

**Issue by wangxa** [nvim-telescope/telescope-frecency.nvim#252]
I just sync the latest submit and when the recent file list function is triggered, the following error occurs:

```
E5108: Error executing lua ...re/nvim/lazy/telescope-frecency.nvim/lua/frecency/
fs.lua:90: attempt to yield across C-call boundary
stack traceback:
        [C]: in function 'fs_stat'
        ...re/nvim/lazy/telescope-frecency.nvim/lua/frecency/fs.lua:90: in funct
ion 'exists'
        ...m/lazy/telescope-frecency.nvim/lua/frecency/database.lua:58: in funct
ion 'filename'
        ...m/lazy/telescope-frecency.nvim/lua/frecency/database.lua:77: in funct
ion 'start'
        ...nvim/lazy/telescope-frecency.nvim/lua/frecency/klass.lua:31: in funct
ion 'init'
        ...nvim/lazy/telescope-frecency.nvim/lua/frecency/klass.lua:41: in funct
ion 'setup'
        .../nvim/lazy/telescope-frecency.nvim/lua/frecency/init.lua:24: in funct
ion 'frecency'
        ...onfig/nvim/lua/user/plugins/config/telescope/sources.lua:76: in funct
ion 'frecency'
        [string ":lua"]:1: in main chunk
```

It seems the `fs.exists(path)` defined in `fs.lua` is an async funcion, because it directly returns `async.uv.fs_stat(path)`, which is an async function. But when calling, the context is not async. The similar issue also occurs at `async.utils.sleeep()` in `database/table.lua`.

**Comment by delphinus** [comment]
@wangxa Thank you for report. Can you use #254 to confirm this issue is solved?

**Comment by wangxa** [comment]
Thank you. It works. 

But the icons for some file type, like JSON are missing, and also the highlight group for icons and  file paths do not work.

**Comment by delphinus** [comment]
Thanks for confirmation. The problem of icon colors should already be fixed by #253 ;)
